internal/ingress/annotations/authtls/main.go

```diff
@@ -130,9 +130,14 @@ func (a authTLS) Parse(ing *networking.Ingress) (interface{}, error) {
 		config.PassCertToUpstream = false
 	}
 
-	config.MatchCN, err = parser.GetStringAnnotation("auth-tls-match-cn", ing)
-	if err != nil || !commonNameRegex.MatchString(config.MatchCN) {
-		config.MatchCN = ""
+	config.ExactMatchCN, err = parser.GetStringAnnotation("auth-tls-exact-match-cn", ing)
+	if err != nil {
+		config.ExactMatchCN = ""
+	}
+
+	config.RegexMatchCN, err = parser.GetStringAnnotation("auth-tls-regex-match-cn", ing)
```

Comment:
I want actually that we validate if this is a valid regex per NGINX and not just add it to nginx.conf.

Based on the feedbacks, do we have a common pattern of required strings, like " REGEX = domain.*" or something like that, in a way we can sanitize/remove possible dangerous characters?

If so, let's do this (in both fields actually, we should validate what character set is allowed here)

Reply:
hmm yea we can do this, I think it will always be like you say something like domain.*, but for example say the person wanted to use a regex match and the CN was = my.example.com and their regex only contained "example", would it fail because it doesn't have a dot after?

Or what if the domain contains numbers? Any suggestions on what a good pattern could look like here? probably something that contains letters, numbers or dots, but no other characters (for example, brackets "<", ">" ?